test/integration/acoustic_comms.cc

```diff
@@ -0,0 +1,155 @@
+#include "../helpers/EnvTestFixture.hh"
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+class AcousticCommsTest : public InternalFixture<::testing::Test>
+{
+  public:
+    void CommsTestFixture(
+        std::string worldFile,
+        std::string srcAddr, std::string dstAddr,
+        int numMsgs = 3)
+    {
+      // Start server
+      ServerConfig serverConfig;
+      const auto sdfFile =
+        gz::common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+          "examples", "worlds", worldFile);
+      serverConfig.SetSdfFile(sdfFile);
+
+      Server server(serverConfig);
+      EXPECT_FALSE(server.Running());
+      EXPECT_FALSE(*server.Running(0));
+
+      // Run server
+      size_t iters = 1000;
+      server.Run(true, iters, false);
+
+      int msgCounter = 0;
+      std::mutex mutex;
+      auto cb = [&](const msgs::Dataframe &_msg) -> void
+      {
+        // Verify msg content
+        std::lock_guard<std::mutex> lock(mutex);
+        std::string expected = "hello world " + std::to_string(msgCounter);
+
+        gz::msgs::StringMsg receivedMsg;
+        receivedMsg.ParseFromString(_msg.data());
+        EXPECT_EQ(expected, receivedMsg.data());
+        msgCounter++;
+      };
+
+      // Create subscriber.
+      gz::transport::Node node;
+      std::string addr  = dstAddr;
+      std::string subscriptionTopic = dstAddr + "/rx";
+
+      // Subscribe to a topic by registering a callback.
+      auto cbFunc = std::function<void(const msgs::Dataframe &)>(cb);
+      EXPECT_TRUE(node.Subscribe(subscriptionTopic, cbFunc))
+          << "Error subscribing to topic [" << subscriptionTopic << "]";
+
+      // Create publisher.
+      std::string publicationTopic = "/broker/msgs";
+      auto pub = node.Advertise<gz::msgs::Dataframe>(publicationTopic);
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      // Prepare the message.
+      gz::msgs::Dataframe msg;
+      msg.set_src_address(srcAddr);
+      msg.set_dst_address(addr);
+
+      // Publish some messages.
+      gz::msgs::StringMsg payload;
+      int pubCount = numMsgs;
+      for (int i = 0; i < pubCount; ++i)
+      {
+        // Prepare the payload.
+        payload.set_data("hello world " + std::to_string(i));
+        std::string serializedData;
+        EXPECT_TRUE(payload.SerializeToString(&serializedData))
+            << payload.DebugString();
+        msg.set_data(serializedData);
+        EXPECT_TRUE(pub.Publish(msg));
+        server.Run(true, 100, false);
+      }
+
+      // Verify subscriber received all msgs.
+      int sleep = 0;
+      bool done = false;
+      while (!done && sleep++ < 3)
+      {
+        {
+          std::lock_guard<std::mutex> lock(mutex);
+          done = (msgCounter == pubCount) && (pubCount != 0);
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      }
+      EXPECT_EQ(pubCount, msgCounter);
+    }
```

Comment:
@adityapande-1995 nit: consider
```suggestion
      bool done = false;
      for (int sleep = 0; !done && sleep < 3; ++sleep)
      {
        std::this_thread::sleep_for(std::chrono::milliseconds(100));
        {
          std::lock_guard<std::mutex> lock(mutex);
          done = (msgCounter == pubCount) && (pubCount != 0);
        }
      }
      EXPECT_TRUE(done);
      EXPECT_EQ(pubCount, msgCounter);
    }
```

Reply:
Done https://github.com/gazebosim/gz-sim/pull/1608/commits/d99fc281e15767b744f58f01847e757076a6d042